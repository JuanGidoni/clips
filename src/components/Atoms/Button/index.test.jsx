import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./index";

describe("<Button /> testing suits", () => {
  const mockClick = jest.fn();
  test("check button snapshot", () => {
    const { container } = render(
      <Button
        text="Testing"
        click={mockClick}
        className="btn"
        testId="buttonTest"
      />
    );
    expect(container).toMatchSnapshot();
  });
  test("should render with props", () => {
    render(
      <Button
        text="Testing"
        click={mockClick}
        className="btn"
        testId="buttonTest"
      />
    );
    const button = screen.getByTestId("buttonTest");
    fireEvent.click(button);
    expect(mockClick).toHaveBeenCalledTimes(1);
    expect(button.innerHTML).toBe("Testing");
    expect(button).toHaveClass("btn");
    expect(button).toHaveAttribute("data-testid", "buttonTest");
    expect(button).toBeInTheDocument();
  });
});
