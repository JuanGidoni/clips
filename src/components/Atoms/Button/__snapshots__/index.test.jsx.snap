// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> testing suits check button snapshot 1`] = `
<div>
  <button
    class="btn"
    data-testid="buttonTest"
  >
    Testing
  </button>
</div>
`;
